library/manageiq_alert.py

```diff
@@ -0,0 +1,282 @@
+
+    def alert_update_required(self, alert_id, description, expression, miq_entity, options, enabled):
+        """ Returns true if the expression, miq_entity, options, or enabled passed for
+            the alert differ from the alert's existing ones, False otherwise.
+        """
+        url = "{api_url}/alert_definitions/{alert_id}".format(api_url=self.api_url, alert_id=alert_id)
+        try:
+            result = self.client.get(url)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to get alert {description} details. Error: {error}".format(description=description, error=e))
+
+        # remove None values from expression and options dicts
+        current_expression = dict((k, v) for k, v in result['expression']['exp'].items() if v is not None)
+        current_options = dict((k, v) for k, v in result['options'].items() if v is not None)
+
+        attributes_tuples = [(current_expression, expression), (result['db'], miq_entity), (current_options, options), (result['enabled'], enabled)]
+        for tup in attributes_tuples:
+            if tup[1] is not None and tup[0] != tup[1]:
+                return True
+        return False
+
+    def update_alert_if_required(self, alert_id, description, expression, miq_entity, options, enabled):
+        """Updates the alert in manageiq.
+
+        Returns:
+            Whether or not a change took place and a message describing the
+            operation executed.
+        """
+        if not self.alert_update_required(alert_id, description, expression, miq_entity, options, enabled):
+            return dict(
+                changed=self.changed,
+                msg="Alert {description} already exist, no need for updates".format(description=description))
+
+        url = '{api_url}/alert_definitions/{alert_id}'.format(api_url=self.api_url, alert_id=alert_id)
+        resource = {'description': description, 'expression': expression, 'db': miq_entity,
+                    'options': options, 'enabled': enabled}
+        try:
+            result = self.client.post(url, action='edit', resource=resource)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update alert {description}: {error}".format(description=description, error=e))
+        self.changed = True
+        return dict(
+            changed=self.changed,
+            msg="Successfully updated alert {description}: {alert_details}".format(description=description, alert_details=result))
+
+    def create_alert(self, description, expression, miq_entity, options, enabled):
+        """Creates the alert in manageiq.
+
+        Returns:
+            Whether or not a change took place and a message describing the
+            operation executed.
+        """
+        url = '{api_url}/alert_definitions/'.format(api_url=self.api_url)
+        resource = {'description': description, 'expression': expression,
+                    'db': miq_entity, 'options': options, 'enabled': enabled}
```

Comment:
you must pass some params here as ruby symbols (:x) or this will not work - at least not with the example provided.